src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -108,22 +108,22 @@ function initial_condition_weak_blast_wave(x, t, equations::IdealMhdMultiIonEqua
   phi = atan(y_norm, x_norm)
 
   # Calculate primitive variables
-  rho = zero(real(equations))
+  rho = 0
   if r > 0.5
     rho = 1.0
   else
     rho = 1.1691
   end
-  v1 = r > 0.5 ? 0.0 : 0.1882 * cos(phi)
-  v2 = r > 0.5 ? 0.0 : 0.1882 * sin(phi)
+  v1 = r > 0.5 ? 0 : 0.1882 * cos(phi)
+  v2 = r > 0.5 ? 0 : 0.1882 * sin(phi)
```

Comment:
```suggestion
  v1 = r > 0.5 ? 0.0 : 0.1882 * cos(phi)
  v2 = r > 0.5 ? 0.0 : 0.1882 * sin(phi)
```
Also here you should keep the `0.0`, as otherwise this code bcomes type unstable (v1 could be an `Int` or ` Float64`.

Reply:
Thanks, @sloede. I think I got it. We should use `0` when the variable/array was already "declared", i.e., when there's no ambiguity about the variable type. We should use `0.0` or `zero(variable_with_right_type)` when there might be ambiguity.

Reply:
Yes, exactly. We use `zero(variable_with_right_type)` only in our generic code, e.g., when we initialize our `Equations` struct or when we create a temporary array inside a `flux` function etc.

In the initial conditions it is not necessary: Here you have a concrete problem that you want to compute with float values, so there's no need to be more generic than necessary.